pandas/io/gbq.py

```diff
@@ -477,11 +477,17 @@ def verify_schema(self, dataset_id, table_id, schema):
         from apiclient.errors import HttpError
 
         try:
-            return (self.service.tables().get(
+            original_schema = self.service.tables().get(
                 projectId=self.project_id,
                 datasetId=dataset_id,
                 tableId=table_id
-            ).execute()['schema']) == schema
+            ).execute()['schema']
+
+            modified_schema_field_list = [
+                {key: field[key] for key in field if key != 'description'}
```

Comment:
Sure! You could argue that [test_df_with_matching_schema_should_verify_schema](https://github.com/medullaskyline/pandas/blob/fix-to_gbq/pandas/io/tests/test_gbq.py#L355) confirms that `'description'` is ignored, but I see your point.

So I added [test_verify_schema_should_ignore_column_description](https://github.com/medullaskyline/pandas/blob/fix-to_gbq/pandas/io/tests/test_gbq.py#L370). The idea was to first prove that `'description'` was the one and only key in the BigQuery table's schema that doesn't exist in the schema generated from the DataFrame. Then if the generated schema is verified with `verify_schema`, that proves `description` is in fact ignored. Feedback welcome.

There is one test that fails but that's because of this commit in [testing.assert_index_equal](https://github.com/pydata/pandas/commit/c76ca444fd42675982e81394736d941e2f2629db#diff-9d7323cc6b0cfa3f4482bc8d4268966dR748).